shogun-lib/src/main/java/de/terrestris/shogun/lib/solr/ShapeImporter.java

```diff
@@ -0,0 +1,137 @@
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.terrestris.shogun.lib.solr;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.Http2SolrClient;
+import org.apache.solr.client.solrj.impl.XMLResponseParser;
+import org.apache.solr.common.SolrInputDocument;
+import org.geotools.data.DataStore;
+import org.geotools.data.DataStoreFinder;
+import org.geotools.data.FeatureSource;
+import org.geotools.data.store.ReprojectingFeatureCollection;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.geometry.jts.WKTWriter2;
+import org.geotools.referencing.CRS;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Polygon;
+import org.opengis.feature.Property;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.filter.Filter;
+import org.opengis.referencing.FactoryException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Import a shape file's fields into solr. Usage:
+ * <code>
+ *     ShapeImporter importer = new ShapeImporter("http://shogun-solr:8983/solr/search/", "myshape.shp", "myshape", "municipalities");
+ *     importer.importShape();
+ * </code>
+ * This will import the features from the shape file into solr setting the special field 'category' to 'municipalities'
+ * and using 'myshape_' as a prefix for the ids. The solr core 'search' will be used.
+ */
+@Log4j2
+public class ShapeImporter {
+
+    private final String solrUrl;
+    private final String file;
+    private final String prefix;
+    private final String category;
+
+    public ShapeImporter(String solrUrl, String file, String prefix, String category) {
+        this.solrUrl = solrUrl;
+        this.file = file;
+        this.prefix = prefix;
+        this.category = category;
+    }
+
+    private SolrInputDocument convertFromFeature(SimpleFeature feature) {
+        SolrInputDocument doc = new SolrInputDocument();
+        WKTWriter2 wkt = new WKTWriter2();
+        doc.addField("id", String.format("%s_%s", prefix, feature.getID()));
+        doc.addField("category", category);
+        doc.addField("title", feature.getProperty("EBENE").getValue() + " " + feature.getProperty("NAME").getValue());
```

Comment:
```suggestion
```
Maybe remove this line as we found out that these fields are specific fields from another project?